client/tests/test_widget.py

```diff
@@ -0,0 +1,135 @@
+from qiskit.quantum_info.random import random_pauli
+from purplecaffeine.core import LocalStorage, Trial
+from purplecaffeine.widget import Widget
+
+
+class TestWidget(TestCase):
+    """TestTrial."""
+
+    def setUp(self) -> None:
+        """SetUp local storage backend for widget use"""
+        curr_dir = os.path.dirname(os.path.abspath(__file__))
+        self.save_path = os.path.join(curr_dir, "test")
+        if not os.path.exists(self.save_path):
+            Path(self.save_path).mkdir(parents=True, exist_ok=True)
+        self.local_storage = LocalStorage(path=self.save_path)
+
+    def test_render_pagination(self):
+        """Test to check that pagination buttons work correctly"""
+        self.add_n_trials(20)
+        widget = Widget(self.local_storage)
+        box = widget.render_pagination()
+        previous_button = box.children[0]
+        next_button = box.children[1]
+        self.assertFalse(next_button.disabled)
+        self.assertTrue(previous_button.disabled)
+
+    def test_render_trial(self):
+        """Test to check that trails buttons are being displayed"""
+        self.add_n_trials(5)
+        widget = Widget(self.local_storage)
+        tabs = widget.render_trial()
+        info = tabs.children[0]
+        self.assertTrue(self.local_storage.list()[0].uuid in info.value)
+
+    def test_search(self):
+        """Test to check the search function"""
+        self.add_n_trials(20)
+        widget = Widget(self.local_storage)
+        box = widget.search()
+        search_button = box.children[0]
+        input_text = box.children[1]
+        input_text.value = self.local_storage.list()[0].name
+        search_button.click()
+        self.assertEqual(widget.search_value, self.local_storage.list()[0].name)
+        self.assertEqual(len(widget.current_page_trials), 1)
+        input_text.value = ""
+        search_button.click()
+        self.assertEqual(len(widget.current_page_trials), 10)
+
+    def test_load_detail(self):
+        """Test to check that the selected trial is the first one"""
+        self.add_n_trials(2)
+        widget = Widget(self.local_storage)
+        first_trail = self.local_storage.list()[0]
+        button = widgets.Button(
+            description=first_trail.name,
+            disabled=False,
+            button_style="",
+            tooltip=first_trail.uuid,
+            icon="",
+        )
+        widget.load_detail(button)
+        self.assertEqual(widget.selected_trial.uuid, first_trail.uuid)
+
+    def test_trial_list(self):
+        """Test to check that new trials are being added."""
+        self.add_n_trials(20)
+        widget = Widget(self.local_storage)
+        self.assertEqual(
+            widget.render_trails_list().children[0].tooltip,
+            self.local_storage.list()[0].uuid,
+        )
+        self.assertEqual(
+            widget.render_trails_list().children[0].tooltip, widget.selected_trial.uuid
+        )
+
+    def add_n_trials(self, number_of_trials):
```

Comment:
Move the function on top, for better reading :)

Reply:
I'm too lazy :) next time :)